server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java

```diff
@@ -501,4 +502,41 @@ public void testNodeProcessorsFloatValidation() {
         }
     }
 
+    // This test must complete to ensure that our basic infrastructure is working as expected.
+    // Specifically that ExecutorScalingQueue, which subclasses LinkedTransferQueue, correctly
+    // tracks tasks submitted to the executor.
+    public void testLTQIssue() {
+        final var executorService = EsExecutors.newScaling(
+            "test",
+            0,
+            1,
+            60,
+            TimeUnit.SECONDS,
+            true,
+            EsExecutors.daemonThreadFactory("test"),
+            new ThreadContext(Settings.EMPTY)
+        );
+        try {
+            final var countDownLatch = new CountDownLatch(10);
```

Comment:
The actual count shouldn't matter here either:

```suggestion
            final var countDownLatch = new CountDownLatch(between(1, 10));
```